datastore/multi.go

```diff
@@ -35,14 +37,15 @@ type multistore struct {
 var _ MultiStore = (*multistore)(nil)
 
 // MultiStoreFrom creates a MultiStore from a root datastore.
-func MultiStoreFrom(rootstore DSReaderWriter) MultiStore {
-	block := prefix(rootstore, blockStoreKey)
+func MultiStoreFrom(rootstore ds.Datastore) MultiStore {
+	rootRW := AsDSReaderWriter(rootstore)
 	ms := &multistore{
-		root:   rootstore,
-		data:   prefix(rootstore, dataStoreKey),
-		head:   prefix(rootstore, headStoreKey),
-		system: prefix(rootstore, systemStoreKey),
-		dag:    NewDAGStore(block),
+		root:   rootRW,
+		data:   prefix(rootRW, dataStoreKey),
+		head:   prefix(rootRW, headStoreKey),
+		peer:   namespace.Wrap(rootstore, rootStoreKey),
```

Comment:
question: Why is the peerstore not prefixed like the rest of the stores?

Reply:
I pushed the documentation of this at the same time you commented.

Reply:
:grin: I thought I had gone mad/blind when I refreshed to add another comment and saw it lol.

question: Why do we want to rely on the libp2p peerstore to assign the prefix?

Reply:
Because it's hard coded in the libp2p pstoreds package. If we assign a prefix of `peers` here we end up with what we had before `/db/peers/peers/addr/...`

Reply:
That's a fair observation. What if we make it specific? `db/ps/peers/...`

Reply:
Totally happy with that :) I know it seems unlikely to ever cause problems, but it feels a bit wrong to rely on just giving it the root.

Reply:
✅ done